### PR TITLE
[TIMELINE] Introduce Manifesto entity (& translatable)

### PR DESCRIPTION
**Pull request by Remg**

### CODE REVIEW

File: src/Entity/Timeline/Manifesto.php [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        return $this->id;
+    }
+
+    /**
+     * @Algolia\Attribute
+     */
+    public function titles(): array
+    {
+        /** @var ProfileTranslation $french */
+        if (!$french = $this->translate('fr')) {
```
Comment (nicoweb) [inline src/Entity/Timeline/Manifesto.php]: maybe you can put this code in `__construct()` method and create properties `$french` and `$english` in order to no repeat it in `slugs()` method

Comment (Remg) [reply src/Entity/Timeline/Manifesto.php]: made a generic method in the abstract class :+1:

File: src/Admin/TimelineProfileAdmin.php [hunk 2]
```diff
@@ -53,8 +54,10 @@ protected function configureDatagridFilters(DatagridMapper $datagridMapper)
     protected function configureListFields(ListMapper $listMapper)
     {
         $listMapper
-            ->addIdentifier('title', null, [
+            ->addIdentifier('title', TextType::class, [
```
Comment (ottaviano) [inline src/Admin/TimelineProfileAdmin.php]: not need this change for ListAction

File: src/Entity/Timeline/Measure.php [hunk 3]
```diff
@@ -312,22 +332,17 @@ public function profileIds(): array
     /**
      * @Algolia\Attribute
      */
-    public function titles(): array
+    public function manifestoId(): ?int
```
Comment (ottaviano) [inline src/Entity/Timeline/Measure.php]: maybe `getManifestoId()` with `@Algolia\Attribute(algoliaName="...")` if you want change the attribute name?

File: src/Entity/Timeline/Theme.php [hunk 4]
```diff
@@ -164,64 +164,45 @@ public function profileIds(): array
     /**
      * @Algolia\Attribute
      */
-    public function titles(): array
+    public function manifestoIds(): array
```
Comment (ottaviano) [inline src/Entity/Timeline/Theme.php]: same for the method name

File: src/Entity/Timeline/Theme.php [hunk 5]
```diff
@@ -164,64 +164,45 @@ public function profileIds(): array
...
-        /** @var ThemeTranslation $french */
-        if (!$french = $this->translate('fr')) {
-            return [];
-        }
+        $manifestos = new ArrayCollection();
 
-        /** @var ThemeTranslation $english */
-        if (!$english = $this->translate('en')) {
-            $english = $french;
+        foreach ($this->measures as $measure) {
```
Comment (ottaviano) [inline src/Entity/Timeline/Theme.php]: what about: 
```php

return array_unique(
    array_map(
        function (Measure $m) { return $m->getManifesto()->getId();}, 
        $measures
    )
);
```

File: src/Admin/TimelineManifestoAdmin.php [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+                'label' => 'Description',
+                'show_filter' => true,
+            ])
+        ;
+    }
+
+    protected function configureListFields(ListMapper $listMapper)
+    {
+        $listMapper
+            ->addIdentifier('title', TextType::class, [
```
Comment (ottaviano) [inline src/Admin/TimelineManifestoAdmin.php]: `TextType` => `null`

File: src/Admin/TimelineManifestoAdmin.php [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+                'label' => 'Titre',
+                'virtual_field' => true,
+                'template' => 'admin/timeline/theme/list_title.html.twig',
```
Comment (ottaviano) [inline src/Admin/TimelineManifestoAdmin.php]: could you try to use `translate.title` as the path of property to display and avoid using the custom template?

Comment (Remg) [reply src/Admin/TimelineManifestoAdmin.php]: doesnt work as it shows up the english translations

File: src/Entity/Timeline/Manifesto.php [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+     * @Algolia\Attribute(algoliaName="descriptions")
+     */
+    public function getDescriptions(): array
+    {
+        return $this->getFieldTranslations('description');
+    }
+
+    public function exportTitles(): string
+    {
+        return join(', ', $this->getTitles());
```
Comment (ktherage) [inline src/Entity/Timeline/Manifesto.php]: What about using `implode` directly as `join` is his alias (https://www.php.net/manual/en/function.join.php) ?

**Pull request merged**